src/format.ts

```diff
@@ -110,8 +126,137 @@ function checkFormat(srcFiles: string[], baseArgs: string[]): Promise<boolean> {
     out.on('data', (data: Buffer) => {
       output += data;
     });
-    out.on('end', () => {
+
+    out.on('end', async () => {
+      const files: string[] =
+          output.split('<?xml version=\'1.0\'?>\n').slice(1);
+      for (let i = 0; i < files.length; i++) {
+        if (files[i].indexOf('<replacement ') === -1) {
+          continue;
+        }
+        const replacements = getReplacements(files[i]);
+        if (replacements.length > 0) {
+          const diff = await getDiffObj(srcFiles[i], replacements);
+          printDiffs(diff, options);
+        }
+      }
       resolve(output.indexOf('<replacement ') === -1 ? true : false);
     });
   });
 }
+
+/**
+ * Parses through xml string for the replacement string, offset, length and
+ * returns all of the necessary replacements for the file
+ *
+ * @param output xml string from clangFormat
+ */
+function getReplacements(fileXML: string): Replacement[] {
+  const replacements: Replacement[] = [];
+
+  let xmlLines = fileXML.split('\n');
+  // The first and last two elements in xmlLines are not needed
+  xmlLines = xmlLines.slice(1, xmlLines.length - 2);
```

Comment:
Suggestion: It wasn't immediately obvious to me that `xmlLines.length - 2` is correct. If you define `xmlLines` as `fileXML.trim().split('\n')` then it will guarantee one element per non-empty line, so you could probably use `xmlLines.length - 1`.

Reply:
done